src/cli.ts

```diff
@@ -327,6 +304,55 @@ async function main() {
     }
   }
 
+  if (!currentVersion) {
+    const manifestPath = projectRootDirectory
+      ? path.join(projectRootDirectory, 'package.json')
+      : path.resolve('package.json');
+
+    try {
+      const manifestText = await fs.readFile(manifestPath, {
+        encoding: 'utf-8',
+      });
+      const manifest = JSON.parse(manifestText);
+      currentVersion = manifest.version;
+    } catch (error) {
+      if (error.code === 'ENOENT') {
+        exitWithError(
+          `Package manifest not found at path: '${manifestPath}'\nRun this script from the project root directory, or set the project directory using the '--root' flag.`,
+        );
+        return;
+      } else if (error.code === 'EACCES') {
+        exitWithError(
+          `Access to package manifest is forbidden by file access permissions: '${manifestPath}'`,
+        );
+        return;
+      } else if (error.name === 'SyntaxError') {
+        exitWithError(
+          `Package manifest cannot be parsed as JSON: '${manifestPath}'`,
+        );
+        return;
+      }
+    }
+  }
+
+  if (!currentVersion) {
+    exitWithError(
+      `Version not found. Please set the --currentVersion flag, or run this as an npm script from a project with the 'version' field set.`,
+    );
+    return;
+  } else if (currentVersion && semver.valid(currentVersion) === null) {
+    exitWithError(`Current version is not valid SemVer: '${currentVersion}'`);
+    return;
+  } else if (!repoUrl) {
+    exitWithError(
+      `npm package repository URL not found. Please set the '--repo' flag, or run this as an npm script from a project with the 'repository' field set.`,
+    );
+    return;
+  } else if (!isValidUrl(repoUrl)) {
+    exitWithError(`Invalid repo URL: '${repoUrl}'`);
+    return;
+  }
```

Comment:
```suggestion
    return exitWithError(
      `Version not found. Please set the --currentVersion flag, or run this as an npm script from a project with the 'version' field set.`,
    );
  }
  if (semver.valid(currentVersion) === null) {
    return exitWithError(`Current version is not valid SemVer: '${currentVersion}'`);
    return;
  }
  if (!repoUrl) {
    exitWithError(
      `npm package repository URL not found. Please set the '--repo' flag, or run this as an npm script from a project with the 'repository' field set.`,
    );
    return;
  }
  if (!isValidUrl(repoUrl)) {
    return exitWithError(`Invalid repo URL: '${repoUrl}'`);
  }
```

Reply:
The pattern of placing the return on the following line was to make complying with the ESLint "consistent-return" rule easier.

Now that you're pointing it out, I think I agree that this style is harder to read and more error prone. I can adjust that in a follow-up PR and find a better solution for complying with that lint rule. I'd rather avoid doing it here because the same pattern is used in more places, and it's distinct from the main purpose of this PR.

Reply:
Done in #122